### DISCUSSION TITLE
How can I test FileUpload ?

### DISCUSSION THREAD

**Discussion by studio-creativo** [filamentphp/filament#11963] [Help]
### Package

Form builder

### Package Version

v2.0.0

### How can we help you?

Hi, I'm trying to test the validation for uploaded file. The rules are: required and dimension min:800px.

Below is my code, but I always receive this error. 

**League\Flysystem\FileNotFoundException: File not found at path: tmp-for-tests**

`public function test_if_files_validation_works()
    {
        $this->actingAs($this->createAdmin());

        Storage::fake('tmp-for-tests');
        $file = UploadedFile::fake()->image('avatar.png',500);

        Livewire::test(CreateImage::class)
        ->fillForm([
            'description' => $this->faker()->text(10),
            'fonte_type' => 'test',
            'tags' => [$this->faker()->text(10),$this->faker()->text(10)],
            'filename' => $file
        ])
        ->call('submit')
        ->assertHasFormErrors(['filename']);
        }`

Can anyone help me to solve this problem?

**Comment by leandrocfe** [comment]
This will work if you use `preserveFileNames()` in the Fileupload

```php
FileUpload::make('attachment')
    ->preserveFilenames()
```

https://github.com/filamentphp/filament/discussions/9575#discussioncomment-7532389

**Reply by studio-creativo** [reply]
Hi thank you for your response but this isn't working in my case. Nothing changed.
Have you any other advice?
